omod/src/main/java/org/openmrs/module/reportingrest/web/resource/MappedReportDefinitionConverter.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ * <p>
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ * <p>
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.reportingrest.web.resource;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.openmrs.module.reporting.report.renderer.RenderingMode;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Converter;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+
+/**
+ * {@link Converter} for {@link RenderingMode}s
+ */
+@Handler(supports = Mapped.class, order = 1)

```

Comment:
I think this is going to handle all Mapped, not just Mapped<ReportDefinition> so I think you may need to generalize this to handle more than just Mapped<ReportDefinition>.

Reply:
@djazayeri do you mean that this class should be extending `Converter<Mapped<T extends Parameterizable>>` rather than being a special case for `ReportDefinition` (aka `Converter<Mapped<ReportDefinition>>`)?

Reply:
@mks-d yes I think that is right.  The converter says it handles Mapped, so we need to handle Mapped.  If we want to code specifically for a Mapped<ReportDefinition> we can do that in a switch and throw an UnsupportedOperationException if it is not a ReportDefinition I suppose, but otherwise we should handle it generically

Reply:
@mseaton @djazayeri The Mapped class is created by the Handler dynamically.  The type associated with the relevant Mapped class will not be available at runtime. So I couldn't make it generalised.